### PR TITLE
[hue] Prevent NPE if 'modelid' is 'null'

### PR DESCRIPTION
**Pull request by cweitkamp**
- Prevent NPE if `modelid` is `null`

When pointing the Hue binding to a deCONZ emulated bridge it may occur that devices using an up-to-date firmware, which is not supported properly, and some properties (e.g. the `modelid`) have a `null` value. I decided to ignore this information because it is not mandatory (see https://community.openhab.org/t/huebridgehandler-and-huelightdiscoveryservice-error-further-auto-discovery-won-t-work/62693/).

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/FullHueObject.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ protected void setType(final String type) {
      *
      * @return model id
      */
-    public String getModelID() {
+    public @Nullable String getModelID() {
         return modelid;
```
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/FullHueObject.java]: So, also the local variable `modelid` is `@Nullable`.
In front of adding `@Nullable` annotation, you should also mark the class `@NonNullByDefault` fist.

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java [hunk 2]
```diff
@@ -144,13 +144,15 @@ private void onLightAddedInternal(FullLight light) {
...
 
-        String modelId = light.getModelID().replaceAll(HueLightHandler.NORMALIZE_ID_REGEX, "_");
+        String modelId = light.getModelID();
 
         if (thingUID != null && thingTypeUID != null) {
             ThingUID bridgeUID = hueBridgeHandler.getThing().getUID();
             Map<String, Object> properties = new HashMap<>();
             properties.put(LIGHT_ID, light.getId());
-            properties.put(Thing.PROPERTY_MODEL_ID, modelId);
+            if (modelId != null) {
```
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java]: If `modelId` is only used if `thingUID != null && thingTypeUID != null` I would move the variable definition and initialization into this branch.

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java [hunk 3]
```diff
@@ -208,13 +210,15 @@ private void onSensorAddedInternal(FullSensor sensor) {
         ThingUID thingUID = getThingUID(sensor);
         ThingTypeUID thingTypeUID = getThingTypeUID(sensor);
 
-        String modelId = sensor.getModelID().replaceAll(HueLightHandler.NORMALIZE_ID_REGEX, "_");
+        String modelId = sensor.getModelID();
```
Comment (maggu2810) [inline extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java]: See above.

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java (hunk 2 above)
Comment (cweitkamp) [reply extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/discovery/HueLightDiscoveryService.java]: No, it is used in the `else` branch too.

File: extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/FullHueObject.java (hunk 1 above)
Comment (cweitkamp) [reply extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/FullHueObject.java]: Done.

**Comment by kaikreuzer** [comment]
> @kaikreuzer Can you please comment if we would like to change the "Hue binding" to work with a deCONZ emulated bridge?

The hue binding even discovers the openHAB hue emulation - so yes, whatever claims to be a hue bridge would be nice to be supported.

**Pull request merged**